components/authz-service/server/v2/migration.go

```diff
@@ -419,133 +420,127 @@ func customPolicyFromV1(pol *storage_v1.Policy) (*storage.Policy, error) {
 		if terms[1] == "*" {
-			return combineTermsIntoResourceSlice([]string{"iam", "introspect"}), nil
+			return "iam:introspect", nil
 		}
 		terms = changeTerm(terms, "auth_introspection", "iam")
 		terms = changeTerm(terms, "introspect_all", "introspect")
 		terms = changeTerm(terms, "introspect_some", "introspect")
-		return combineTermsIntoResourceSlice(terms), nil
+		return combineTermsIntoResource(terms...), nil
 	case "cfgmgmt":
 		return convertV1Cfgmgmt(terms)
 	case "compliance":
 		// Special case
 		if resource == "compliance:profiles:market" {
-			return combineTermsIntoResourceSlice([]string{"compliance", "marketProfiles"}), nil
+			return "compliance:marketProfiles", nil
 		}
-		terms = deleteTerm(terms, "storage")
-		return combineTermsIntoResourceSlice(terms), nil
+		return combineTermsIntoResource(deleteTerm(terms, "storage")...), nil
 	case "events":
 		return convertV1Events(terms)
 	case "ingest":
 		// Special case: "ingest:status" -> "infra:ingest:status" (no wildcards to worry about)
 		if terms[1] == "status" {
-			terms := []string{"infra", "ingest", "status"}
-			return combineTermsIntoResourceSlice(terms), nil
+			return "infra:ingest:status", nil
 		}
 
 		// Special case: "ingest:unified_events" -> "infra:unifiedEvents" (no wildcards to worry about)
 		if terms[1] == "unified_events" {
-			terms := []string{"infra", "unifiedEvents"}
-			return combineTermsIntoResourceSlice(terms), nil
+			return "infra:unifiedEvents", nil
 		}
 
 		terms = changeTerm(terms, "ingest", "infra")
 		terms = changeTerm(terms, "unified_events", "unifiedEvents")
-		return combineTermsIntoResourceSlice(terms), nil
+		return combineTermsIntoResource(terms...), nil
 	case "license":
 		if len(terms) == 1 {
-			return []string{"system:license"}, nil
+			return "system:license", nil
 		}
 		// if len(terms) == 2 aka license:* or license:status
-		return []string{"system:status"}, nil
+		return "system:status", nil
 	case "nodemanagers":
 		// "nodemanagers" -> "infra:nodeManagers"
-		terms[0] = "nodeManagers"
-		terms = append([]string{"infra"}, terms...)
-		return combineTermsIntoResourceSlice(terms), nil
+		return combineTermsIntoResource(prefixTerms("infra", changeTerm(terms, "nodemanagers", "nodeManagers"))...),
+			nil
 	case "nodes":
 		// "nodes" -> "infra:nodes"
-		terms = append([]string{"infra"}, terms...)
-		return combineTermsIntoResourceSlice(terms), nil
+		return combineTermsIntoResource(prefixTerms("infra", terms)...), nil
 	case "secrets":
 		// "secrets" -> "secrets:secrets"
-		terms = append([]string{"secrets"}, terms...)
-		return combineTermsIntoResourceSlice(terms), nil
+		return combineTermsIntoResource(prefixTerms("secrets", terms)...), nil
 	case "telemetry":
 		// either telemetry:config or telemetry:* maps to system:config
-		return combineTermsIntoResourceSlice([]string{"system", "config"}), nil
+		return "system:config", nil
 	case "notifications":
-		return combineTermsIntoResourceSlice(terms), nil
-	case "*":
```

Comment:
This was a bug: `*:foo:bar` would have been converted to `*`. It's now [tighter, above](https://github.com/chef/automate/pull/612/files#diff-a9d9a31aed7e79f7b7936b91698800ceR429)